extras/src/main/java/com/google/gson/typeadapters/RuntimeTypeAdapterFactory.java

```diff
@@ -120,6 +120,18 @@
  *       .registerSubtype(Circle.class)
  *       .registerSubtype(Diamond.class);
  * }</pre>
+ *
+ * <h3>Serialization and deserialization</h3>
+ * In order to serialize and deserialize a polymorphic object,
+ * you must specify the base type explicitly.
+ * <pre>   {@code
+ *   Diamond diamond = new Diamond();
```

Comment:
serialization does not require the type, only deserialization

Reply:
@inder123 hm, there must be something wrong then. Please see the original issue #1440 that raised this PR. Additionally all tests in `RuntimeTypeAdapterFactoryTest.java` specify the type explicitly on serialization.

Reply:
Ah, ok. This is for RuntimeTypeAdapterFactory not core Gson.